internal/appsec/appsec.go

```diff
@@ -135,3 +132,26 @@ func (a *appsec) stop() {
 		a.limiter.Stop()
 	}
 }
+
+func (a *appsec) startRC() {
+	if a.rc == nil {
+		return
+	}
+	defer a.rc.Start()
+	// Set WAF-related RC capabilities and products if the WAF is in good health. We perform this check in order not to
+	// falsely "allow" users to activate ASM through remote config if activation would fail when trying to register a WAF handle
+	// (ex: if the service runs on an unsupported platform).
+	if err := waf.Health(); err != nil {
+		log.Debug("appsec: Remote config: WAF health check failed, WAF-related features will be disabled: %v", err)
+		return
+	}
+	a.rc.Capabilities = append(a.rc.Capabilities, remoteconfig.ASMActivation)
```

Comment:
Avoid it also when the env var is set (no matter the value)

Reply:
1b724a0f8452b2462